src/roiextractors/extractors/inscopixextractors/inscopixsegmentationextractor.py

```diff
@@ -12,51 +13,154 @@ class InscopixSegmentationExtractor(SegmentationExtractor):
     """A segmentation extractor for Inscopix."""
 
     extractor_name = "InscopixSegmentationExtractor"
-    installed = True
-    is_writable = False
-    mode = "file"
-    installation_mesg = ""
 
-    def __init__(self, file_path: PathType):
+    def __init__(self, file_path: PathType, verbose: bool = True):
         """Initialize a InscopixSegmentationExtractor instance.
 
         Main class for extracting segmentation data from Inscopix format.
 
         Parameters
         ----------
-        file_path: str
+        file_path: str or PathType
             The location of the folder containing Inscopix *.mat output file.
+        verbose: bool, default True
+            Whether to print verbose output for warnings and errors.
         """
         if platform.system() == "Darwin" and platform.machine() == "arm64":
             raise ImportError(
-                "The isx package is currently not natively supported on macOS with Apple Silicon. "
-                "Installation instructions can be found at: "
-                "https://github.com/inscopix/pyisx?tab=readme-ov-file#install"
+                "For macOS ARM64, please use a special conda environment setup. " "See README for instructions."
             )
 
         import isx
 
         SegmentationExtractor.__init__(self)
         self.file_path = file_path
-        self.cell_set = isx.CellSet.read(file_path)
+        self.verbose = verbose
+        file_path_str = str(file_path)
+
+        self.cell_set = isx.CellSet.read(file_path_str)
+
+        # Create mappings between original IDs and integer IDs
+        self._original_ids = [self.cell_set.get_cell_name(x) for x in range(self.cell_set.num_cells)]
+        self._id_to_index = {id: i for i, id in enumerate(self._original_ids)}
+        self._index_to_id = {i: id for id, i in self._id_to_index.items()}
+
+        # Cache for metadata to avoid repeated extraction
+        self._metadata_cache = None
 
     def get_num_rois(self) -> int:
         return self.cell_set.num_cells
 
-    def get_roi_image_masks(self, roi_ids=None) -> np.ndarray:
+    def _get_roi_indices(self, roi_ids=None) -> List[int]:
```

Comment:
```suggestion
    def _get_roi_indices(self, roi_ids=None) -> list[int + str]:
```
I think that Pauls prefer to use just list for typing as `List` is just an alias now.